apps/server/src/modules/me/api/dto/me.response.ts

```diff
@@ -0,0 +1,103 @@
+import { ApiProperty } from '@nestjs/swagger';
+
+export class MeAccountResponse {
+	@ApiProperty()
+	id: string;
+
+	constructor(props: MeAccountResponse) {
+		this.id = props.id;
+	}
+}
+
+export class MeSchoolLogoResponse {
+	@ApiProperty({ type: String })
```

Comment:
Wir hatten ja schon im Chat drüber geredet und das ist mMn falsch. Wenn du so den Client generierst, dann ergibt das die Prop `url: string`, was aber falsch ist, weil sie nullable ist. Der richtige Weg ist mMn 
```
@ApiPropertyOptional()
url?: string;
```

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/null
It return key: null in json object which is for me valid for not set string values.

Reply:
@bischofmax es gab dazu glaube ich Gespräche im Client Kontext. Was ist das was ihr dort haben wollt. Key nicht ausgeliefert wenn der Wert undefined ist, oder key vorhanden mit null als Wert.

Reply:
Es stimmt jedenfalls der Typ im generierten API-Client so nicht, weil der dann `string` ist, während `string | null` korrekt wäre. 

Wenn es bestimmte Anforderungen aus dem Client gibt, dann sollten die erstmal grundsätzlich spezifiziert werden, finde ich. Gerade scheint mir das eher auf Halbwissen zu beruhen.

Reply:
Nur mal als Beispiel so sieht der Response vom neuen Schul get Endpunkt aus.
OpenAPI  --> "officialSchoolNumber": "string",
Code --> @ApiPropertyOptional()  officialSchoolNumber?: string;

Reply:
Das ist das, was bei Swagger steht, oder? Da erkennt man die optionalen Props nicht. Guck dir mal den generierten API-Client im Nuxt-Client an. Da ist `officialSchoolNumber` als optional markiert.

Reply:
Ich find die Diskussion um `null` und `undefined` durchaus interessant, aber sie kommt für mich an dieser Stelle sehr überraschend. Normalerweise arbeiten wir ja überall mit optionalen Props und mir war bisher auch kein Problem damit bekannt. Wenn es eins gibt, wie Max das angedeutet hat, dann sollten wir erstmal dieses Problem genauer ausführen, bevor wir was ändern, finde ich.

Reply:
So richtig klar ist mir nicht was jetzt das Problem ist. Möchtest du das ich das wie am Anfang mit Optional und undefined keys zurück setzte? Was genau verbessert das? Eine Entscheidung die für die me Route getroffen wird, hat doch nur Auswirkungen auf genau diese eine Route. Wir sind diejenigen die diese API einbauen werden.

Wobei die Spezifizierung der API Teil des nächsten Tickets ist. Ich hätte das auch leer lassen können, damit wäre das Ticket auch erfüllt. Ich werde nicht in dem Ticket ein OpenAPI Build ausführen und schon mal beispielhaft zu implementieren um zu schauen was mir nun besser gefällt null checks oder key undefined checks.

Reply:
In Absprache mit Sergej hab ichs jetzt auf optionale Props geändert.

Auch wenn der Client Sache des nächsten Tickets ist, hätte ich es falsch gefunden, jetzt hier wissentlich ne API-Spezifikation zu schreiben, die falsche Typen im Client erzeugen wird.